x/bank/README.md

```diff
@@ -236,8 +236,12 @@ accounts. The send keeper does not alter the total supply (mint or burn coins).
 type SendKeeper interface {
     ViewKeeper
 
-    InputOutputCoins(ctx context.Context, inputs types.Input, outputs []types.Output) error
-    SendCoins(ctx context.Context, fromAddr sdk.AccAddress, toAddr sdk.AccAddress, amt sdk.Coins) error
+    AppendSendRestriction(restriction SendRestrictionFn)
+    PrependSendRestriction(restriction SendRestrictionFn)
```

Comment:
The ordering of send restrictions might matter. I expect most things will use `Append`, but there could easily be a situation where a module needs theirs to go before the ones already injected.

I hate when I get into a situation where I just can't do what I know I need to do because the only layer I have access too didn't provide enough control. So, while `Append` will usually be good enough, `Prepend` and `Clear` can easily save the day.

Reply:
> could easily be a situation where a module needs theirs to go before the ones already injected.

But how does the module know whether it's safe to insert its restriction function before all others? That is, I see a problem where modules can be fighting each other for the correct position of their restrictions.

Reply:
The module can't know. The blockchain author can though, and needs to have the tools available to fix ordering problems if the arise.